client/state/posts/selectors.js

```diff
@@ -20,10 +32,40 @@ import { DEFAULT_POST_QUERY } from './constants';
  * @return {Object}          Post object
  */
 export function getPost( state, globalId ) {
-	return state.posts.items[ globalId ];
+	return getPostsByGlobalId( state )[ globalId ];
+}
+
+/**
+ * Returns an array of all posts.
+ *
+ * @param  {Object} state    Global state tree
+ * @return {Array}           Posts array
+ */
+export function getPosts( state ) {
+	return state.posts.items;
 }
 
 /**
+ * Returns a mapping of site ID, post ID pairing to global post ID.
+ *
+ * @param state
+ */
+export const getSitePosts = createSelector(
```

Comment:
> In my mind, this function would return an array of posts for a particular site ID, passed as an argument

How about if we rename this to getPostsBySiteIdAndPostId. This was intended to replace the sitePosts section of the redux tree. Memoized selectors are not intended to be passed arbitrary elements, but we they can be composed in other memoized selectors or used directly in a simple one.

> Perhaps a separate function which calls on the memoized site posts selector (getSitePosts, getPostsBySiteId)?

Exactly:
```javascript
export const getSitePostsBySiteId() = createSelector( 
    getPosts,
    ( posts ) => {
        return groupBy( posts, 'site_ID');
    }
);
export function getSitePosts() {
   return getSitePostsBySiteId( state )[ siteId ];
}

```

> but some of the opinions of the library clash with ours enough that I worry about the confusion it might cause to treat them as the same.

Perhaps if we think of the memoized selectors as things that could live in the redux tree, but happen to be computed. 
A simple wrapper
```
function getSitePosts( state ) {
   return state.posts.sitePosts;
}
```
versus:
```
function getSitePosts( state )  = createSelector( 
   getPosts,
   ( posts ) => {
       //...
   }
}
```
Still has the same usage.